testsuite/integration/elytron-oidc-client/src/test/java/org/wildfly/test/integration/elytron/oidc/client/KeycloakConfiguration.java

```diff
@@ -75,15 +77,22 @@ public static RealmRepresentation getRealmRepresentation(final String realmName,
     }
 
     public static String getAdminAccessToken(String authServerUrl) {
-        return RestAssured
+        RequestSpecification requestSpecification = RestAssured
                 .given()
                 .param("grant_type", "password")
                 .param("username", KeycloakContainer.ADMIN_USER)
                 .param("password", KeycloakContainer.ADMIN_PASSWORD)
-                .param("client_id", "admin-cli")
-                .when()
-                .post(authServerUrl + "/realms/master/protocol/openid-connect/token")
-                .as(AccessTokenResponse.class).getToken();
+                .param("client_id", "admin-cli");
+
+        Response response = requestSpecification.when().post(authServerUrl + "/realms/master/protocol/openid-connect/token");
+
+        while (response.getStatusCode() != 200) {
+            // the Keycloak admin user isn't available yet, keep trying until it is to ensure we can obtain the token
+            // needed to set up the realms for the test
+            response = requestSpecification.when().post(authServerUrl + "/realms/master/protocol/openid-connect/token");
```

Comment:
I feel like we should add some kind of timeout here to be safe in cases where maybe the server never starts.

Reply:
Thanks, James! Will update this soon.

Reply:
@jamezp I've added a timeout now, thanks!